README.md

```diff
@@ -82,7 +84,210 @@ Requirements:
        ```
 
 Build:
-------
+-----------------
+  Aladdin currently supports two ways to build.
+
+  1. Configure with CMake, then build with Make :
+
+       If you are familiar with CMake, or you are interested in a script
+       which can build out-of-source and offer automatical LLVM/Clang
+       installation, choose [CMake](#build-with-cmake)
+
+  2. Directly build with Makefile :
+
+       If you want to keep all the things simple, or you do not have
+       CMake, choose [Makefile](#build-with-makefile)
+
+
+Build with CMake:
+-----------------
+CMake is a configure tool which allows you to out-of-source build.
+Aladdin Requires CMake newer than 2.8.12. By default, CMake
+searches for LLVM 3.4.
+
+
+1. Set `$BOOST_ROOT`,  `TRACER_HOME`, `LLVM_HOME` to let Aladdin know where the tools and libraries are.
+
+   ```
+   export BOOST_ROOT=/your/path/to/boost
+   export TRACER_HOME=/your/path/to/LLVM-Tracer
+   export LLVM_HOME=/your/path/to/llvm
+   ```
+
+2. Use CMake to configure Aladdin and build libraries and executables
+
+   ```
+   mkdir /where/you/want/to/put/your/binary
```

Comment:
supply the following explanation to readme.
```
If you want to put the final executables and libraries on source
directory, pass `-DBUILD_ON_SOURCE=TRUE` arguemnt to `cmake`.
Otherwise, `-DBUILD_ON_SOURCE=FALSE`. The default is FALSE.
```